internal/domain/bonus/service/service.go

```diff
@@ -0,0 +1,108 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	bonus_transactionsModel "github.com/Azzonya/gophermart/internal/domain/bonusTransactions/model"
+	orderModel "github.com/Azzonya/gophermart/internal/domain/order/model"
+	"github.com/Azzonya/gophermart/internal/usecase/bonustransactions"
+	"github.com/Azzonya/gophermart/internal/usecase/order"
+	"sync"
+	"time"
+)
+
+type Service struct {
+	repoAccrual              RepoAccrualI
+	bonusTransactionsService bonustransactions.WithdrawalServiceI
+	orderService             order.OrderServiceI
+	Wg                       sync.WaitGroup
+}
+
+func New(repoAccrual RepoAccrualI, bonusTransactionsService bonustransactions.WithdrawalServiceI, orderService order.OrderServiceI) *Service {
+	return &Service{
+		repoAccrual:              repoAccrual,
+		bonusTransactionsService: bonusTransactionsService,
+		orderService:             orderService,
+	}
+}
+
+func (s *Service) Start(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+
+	go func() {
+		defer ticker.Stop()
+		for range time.Tick(interval) {
+			s.Wg.Add(1)
+			go s.UpdateAccrualInfo(context.Background())
+		}
+	}()
+
+	defer ticker.Stop()
+}
+
+func (s *Service) UpdateAccrualInfo(ctx context.Context) error {
```

Comment:
он точно должен быть экспортируемым? Вызывается только в этом пакете
```suggestion
func (s *Service) updateAccrualInfo(ctx context.Context) error {
```